### PR TITLE
Clement

### PR DESCRIPTION
**Pull request by ClementPinard**
Hi,

as discussed by mails, this is a PR that tries to update the code to work with Ubuntu 18

For now, no feature are added, I just made the following fixes :
 - Change QT4 to QT5
 - Add necessary header and `std::` prefixes to make it build on ubuntu 16 and 18
 - Add OpenGV on CmakeLists, in case it was not installed the traditional way
 - Change README to notify these changes

I also added a DockerFile to build from a blank ubuntu18

I have not added colmap's new camera models, as I have some troubles with the way camera distortion and undistortion are handled (I'll open a discussion in issues), but it will come shortly !

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -315,6 +323,7 @@ target_link_libraries(Test_ICP
   gtest
   gtest_main
 )
+target_compile_options(Test_ICP PRIVATE -g)
```
Comment (puzzlepaint) [inline CMakeLists.txt]: This should probably not be set by default, but rather by CMAKE_BUILD_TYPE?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+RUN apt update && apt install -y git libvtk6-dev libflann-dev libeigen3-dev libboost-all-dev cmake libgmp-dev libglew-dev libgoogle-glog-dev qt5-default libproj-dev libqwt-qt5-dev libpcl-dev
+RUN git clone https://github.com/laurentkneip/opengv.git
+RUN cd opengv \
+  && mkdir build \
+  && cd build \
+  && cmake .. \
+  && make -j8 install \
+  && cd ../../
+
+RUN git clone https://github.com/opencv/opencv.git
```
Comment (puzzlepaint) [inline Dockerfile]: Perhaps one would want to choose some (stable) version here?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+COPY . /dataset-pipeline
+RUN mkdir -p build \
+  && cd build \
+  && rm -r * \
+  && cmake .. \
+  && make -j8
+
+ENV PIPELINE_PATH=/dataset-pipeline/build
+
+#Rests
```
Comment (puzzlepaint) [inline Dockerfile]: #Tests ?

File: src/point_cloud_editor/render_widget.cc [hunk 4]
```diff
@@ -36,6 +36,7 @@
 
 #include "camera/camera_pinhole.h"
 #include "opengl/opengl_util.h"
+#include <Eigen/StdVector>
```
Comment (puzzlepaint) [inline src/point_cloud_editor/render_widget.cc]: Minor: The include order in the codebase should (hopefully) be like this, with a free line between each block:

```c++
// for source files, #include "" of the corresponding header

// system #includes <>

// external library #includes <>

// #includes "" within the same project
```

So, for consistency, the Eigen include should go to the block above.

File: CMakeLists.txt (hunk 1 above)
Comment (ClementPinard) [reply CMakeLists.txt]: Sorry about that, that one slipped through when I solved the memory alignment error that occured with Test_ICP. I'll remove it.

**Comment by ClementPinard** [comment]
Thanks for your insight ! I hopefully adressed all your comments.

Also, I forget to talk about it in the commit comments but I suppressed some warning regarding signed and unsigned comparison. This cleans the building console and eases debugging, but maybe it could be considered as information retention, so I can supress the corresponding files in the CMakeLists.txt if needed.

**Comment by puzzlepaint** [comment]
Thanks! Suppressing that warning seems fine to me.

**Pull request merged**